defaults/main.yml

```diff
@@ -81,8 +81,19 @@ onadata_wsgi_module: "onadata.apps.main.wsgi:application"
 onadata_enable_celery: true
 onadata_celery_app: "onadata.celery"
 onadata_domain: "example.com"
+# Maximum amount of memory a worker can consume before it's replaced
+# by a new worker. Value should be in KiB
+# See: https://docs.celeryproject.org/en/stable/userguide/workers.html#max-memory-per-child-setting
+onadata_celery_worker_max_memory: 195312  # 200MB
 onadata_celeryd_nodes: "{{ onadata_domain }} export-node publish-xls-form-node google-export xls-exports csv-exports kml-exports osm-exports csv-zip-exports sav-zip-exports external-exports zip-exports osm-exports exports"
-onadata_celeryd_opts: "-O fair --concurrency=8 --autoscale=6,1 -Q:{{ onadata_domain }} celery -Q:export-node exports -Q:publish-xls-form-node publish_xlsform -Q:google-export google_export -Q:xls-exports xls_exports -Q:csv-exports csv_exports -Q:kml-exports kml_exports -Q:osm-exports osm-exports -Q:csv-zip-exports csv_zip_exports -Q:sav-zip-exports sav_zip_exports -Q:external-exports external_exports -Q:zip-exports zip_exports -Q:osm-exports osm_exports -Q:exports exports"
+onadata_celeryd_opts: "-O fair --concurrency=8 --autoscale=6,1 -Q:{{ onadata_domain }} celery -Q:export-node exports -Q:publish-xls-form-node publish_xlsform -Q:google-export google_export -Q:xls-exports xls_exports -Q:csv-exports csv_exports -Q:kml-exports kml_exports -Q:osm-exports osm-exports -Q:csv-zip-exports csv_zip_exports -Q:sav-zip-exports sav_zip_exports -Q:external-exports external_exports -Q:zip-exports zip_exports -Q:osm-exports osm_exports -Q:exports exports --max-memory-per-child={{ onadata_celery_worker_max_memory }}"
+# Task hard time limit in seconds. Celery kills & replaces a worker processing a task that exceeds set time
+# See: https://docs.celeryproject.org/en/stable/userguide/configuration.html#std:setting-task_time_limit
+onadata_celery_task_time_limit:
```

Comment:
Hey, was this time limit determined?
It doesnt seem to be defined

Reply:
I don't think it was... I think it's better of having the user define the value for this, _Seemed okay to set the max amount of tasks or memory since those only take effect when a task is completed but this completely kills a task if it goes past a certain time limit which might not be what everyone wants to happen by default... In some cases, you might want stuff like exports/imports to keep going on as long as they aren't memory intensive IMO_

Reply:
> In some cases, you might want stuff like exports/imports to keep going on as long as they aren't memory intensive

Yeah, I agree.
But in the case a task runs upto a day or 2, wouldnt we want to kill that task?
The idea, is to have the worker processing the task killed and replaced with a new one at such a time, ama?
Is this something we may want to leverage?

Reply:
Once again yea it might be nice... but I don't think it's worth a default value here. Seems like something the user of this package should set on their own...

Reply:
You may want to leverage the feature to kill certain tasks after a certain time but in the same case, you may also want tasks to be completed no matter the time they might task. I feel like this option should be left open to the user of the package...

Reply:
no, i see the advantage of setting this as default here. 
Let individual users of the role define this value according to their own system requirements